File: templates/era-python/scripts/viz.py
#!/usr/bin/env python3
import json, os, sys

def main():
    os.makedirs("figures", exist_ok=True)
    src = "runs/primary/metrics.json"
    dst = "figures/figure1.txt"
    if not os.path.exists(src):
        print("Missing metrics; did you run train?", file=sys.stderr)
        return 1
    with open(src) as f:
        data = json.load(f)
    with open(dst, "w") as f:
        f.write(f"Figure1 placeholder: epochs={data.get(epochs)}, lr={data.get(learning_rate)}\\n")
    print(f"Wrote {dst}")
    return 0

if __name__ == "__main__":
    sys.exit(main())

File: templates/era-julia/scripts/viz.jl
#!/usr/bin/env julia
using JSON
import Base.Filesystem: mkpath

function main()
    mkpath("figures")
    if !isfile("runs/primary/metrics.json")
        @warn "Missing metrics; did you run train?"
        return 1
    end
    data = JSON.parsefile("runs/primary/metrics.json")
    open("figures/figure1.txt","w") do io
        write(io, "Figure1 placeholder: epochs=$(data["epochs"])\\n")
    end
    println("Wrote figures/figure1.txt")
    return 0
end

if abspath(PROGRAM_FILE) == @__FILE__
    exit(main())
end
